rust/headless-client/src/standalone.rs

```diff
@@ -209,7 +208,7 @@ pub fn run_only_headless_client() -> Result<()> {
 
         let tun = tun_device.make_tun()?;
         session.set_tun(Box::new(tun));
-        session.set_dns(dns_control::system_resolvers().unwrap_or_default());
+        session.set_dns(dns_control::system_resolvers(dns_control_method).unwrap_or_default());
```

Comment:
Ah I see, you want to reuse the code here. We could still call this function from within the `DnsController` and make the other call-site a bit more ergonomic?

Reply:
`DnsController` does exist there so I made a member function wrapper.
I had to keep the original `system_resolvers` free function because it's used in some place where we won't have a controller at hand.